File: spigot/src/main/java/net/isariel/gaia/spigot/greeting/GreetingManagerImpl.java
package net.isariel.gaia.spigot.greeting;

import net.isariel.lycan.api.service.tablist.TablistService;
import org.bukkit.entity.Player;

import javax.inject.Inject;

/**
 * @author Felix Klauke <fklauke@itemis.de>
 */
public class GreetingManagerImpl implements GreetingManager {

    private final TablistService tablistService;

    @Inject
    public GreetingManagerImpl(TablistService tablistService) {
        this.tablistService = tablistService;
    }

    @Override
    public void greetPlayer(Player player) {
        tablistService.sendHeaderFooter(player, "\nWillkommen auf Isariel\n", "");
    }
}

File: spigot/src/main/java/net/isariel/gaia/spigot/greeting/GreetingManagerImpl.kt
package net.isariel.gaia.spigot.greeting

import net.isariel.lycan.api.service.tablist.TablistService
import org.bukkit.entity.Player

import javax.inject.Inject

/**
 * @author Felix Klauke <info@felix-klauke.de>
 */
class GreetingManagerImpl @Inject
constructor(private val tablistService: TablistService) : GreetingManager {

    override fun greetPlayer(player: Player) {
        tablistService.sendHeaderFooter(player, "\nWillkommen auf Isariel\n", "")
    }
}
